provisioner/terraform/tfparse/funcs.go

```diff
@@ -0,0 +1,162 @@
+package tfparse
+
+import (
+	"github.com/aquasecurity/trivy-iac/pkg/scanners/terraform/parser/funcs"
+	"github.com/hashicorp/hcl/v2/ext/tryfunc"
+	ctyyaml "github.com/zclconf/go-cty-yaml"
+	"github.com/zclconf/go-cty/cty"
+	"github.com/zclconf/go-cty/cty/function"
+	"github.com/zclconf/go-cty/cty/function/stdlib"
+	"golang.org/x/xerrors"
+)
+
+// Functions returns a set of functions that are safe to use in the context of
+// evaluating Terraform expressions without any ability to reference local files.
+// Functions that refer to file operations are replaced with stubs that return a
+// descriptive error to the user.
+func Functions() map[string]function.Function {
+	return allFunctions
+}
+
+var (
+	// Adapted from github.com/aquasecurity/trivy-iac@v0.8.0/pkg/scanners/terraform/parser/functions.go
+	// We cannot support all available functions here, as the result of reading a file will be different
+	// depending on the execution environment.
+	safeFunctions = map[string]function.Function{
+		"abs":             stdlib.AbsoluteFunc,
```

Comment:
Post-merge thought: how do we align Terraform versions?

Reply:
From what I could tell it doesn't look like these functions change all that often.

However, f Terraform adds a new function in future, we'll need to add support for it ourselves. Then again, so will `aquasec/trivy-iac`. This is an unfortunate consequence of Terraform having made the bulk of their internal functions un-importable.

Reply:
Cool, I think we're fine here and we can just update the lib.
My only concern is if they change the _semantics_ of the functions between versions, but I doubt they'd do that.